client/disputes/info/test/index.tsx

```diff
@@ -39,11 +42,17 @@ describe( 'Dispute info', () => {
 			},
 		};
 
-		const { container: info } = render( <Info dispute={ dispute } /> );
+		const { container: info } = render(
+			// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+			// @ts-ignore
```

Comment:
Complains of
- not receiving all props
- dispute doesn't have all the necessary props.

Would it be a proper solution to simply `@ts-ignore` for tests?

Reply:
This is the exact error that I see:

```
Type '{ amount: number; currency: string; created: number; evidence_details: { due_by: number; };
reason: string; status: string; order: { number: string; url: string; }; charge: { id: string; billing_details: { ...; }; }; }'
is missing the following properties from type 'Dispute': id, metadata, productType, evidence
```

Maybe add an `id` prop to `const dispute`, and make metadata, productType, evidence optional props in `Dispute` interface?

Reply:
I dislike the idea of altering definitions for tests. To account for all tests there won't be much of an interface for Dispute left.

Reply:
Hmm, I see your point. Sure, let's go ahead with @ts-ignore 👍

Reply:
Wow, this is ingenius → hover over @ts-ignore to find out why 😆

Reply:
Folks, can you ping me tomorrow to look into this? Thanks!

Reply:
@kalessil pinging you

Reply:
Following feedback from TS linter, I came up with this:

```javascript
	test( 'renders correctly', () => {
		const dispute = {
			status: 'needs_response',
			id: '',
			evidence_details: {
				has_evidence: true,
				due_by: 1573199200,
			},
			metadata: {},
			productType: '',
			order: {
				number: '1',
				url: 'http://test.local/order/1',
			} as unknown,
			evidence: {} as Evidence,
			fileSize: {},
			reason: 'fraudulent',
			charge: {
				id: 'ch_mock',
				billing_details: {
					name: 'Mock customer',
				},
			},
			amount: 1000,
			currency: 'usd',
			created: 1572590800,
		};

		const { container: info } = render(
			<Info isLoading={ false } dispute={ dispute as Dispute } />
		);
		expect( info ).toMatchSnapshot();
	} );

	test( 'renders loading state', () => {
		const { container: info } = render(
			<Info isLoading={ true } dispute={ {} as Dispute } />
		);
		expect( info ).toMatchSnapshot();
	} );
```

Reply:
It asked for `const dispute` to match the definitions and for `render(` parts to supply all required properties.